tests/test_cases/test_cocotb/common.py

```diff
@@ -31,17 +30,15 @@ def _check_traceback(running_coro, exc_type, pattern):
     except exc_type:
         tb_text = traceback.format_exc()
     else:
-        raise TestFailure("Exception was not raised")
+        assert False, "Exception was not raised"
 
     if not re.match(pattern, tb_text):
-        raise TestFailure(
-            (
-                "Traceback didn't match - got:\n\n"
-                "{}\n"
-                "which did not match the pattern:\n\n"
-                "{}"
-            ).format(tb_text, pattern)
-        )
+        assert False, (
+            "Traceback didn't match - got:\n\n"
+            "{}\n"
+            "which did not match the pattern:\n\n"
+            "{}"
+        ).format(tb_text, pattern)
```

Comment:
You could put the `re.match` in the `assert` so we can see what was matched when the assertion rewriting comes in?

Reply:
Oh, yes, indeed.